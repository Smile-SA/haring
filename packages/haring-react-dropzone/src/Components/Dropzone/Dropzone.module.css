.buttonPlus {
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: center;
  margin: auto;
}

.cardFile {
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  height: 70px;
  margin: 10px;
  padding: 10px;
  position: relative;
  width: 80px;

  /*noinspection CssInvalidAtRule*/
  @mixin light {
    background: var(--mantine-color-gray-2);
  }

  /*noinspection CssInvalidAtRule*/
  @mixin dark {
    background: var(--mantine-color-black);
  }
}

.cardFileButtonClose {
  position: absolute;
  right: -7px;
  top: -7px;
}

.cardFileText {
  display: inline-block;
  font-size: 12px;
  height: fit-content;
  margin: auto;
  width: 100%;
}

.cardFileText:first-of-type {
  font-weight: 600;
}

.cardsFile {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.dropzoneBrowse {
  align-items: center;
  display: flex;
  font-weight: 600;
  justify-content: center;
  margin-top: 0px;
}

.dropzoneInner {
  margin: auto;
  pointer-events: auto;
  text-align: center;
}

.dropzoneInner p {
  margin: 10px;
}

.dropzoneRoot {
  display: flex;
  min-height: 100%;
  min-width: 100%;
}

.eye {
  margin-right: 8px;
}
