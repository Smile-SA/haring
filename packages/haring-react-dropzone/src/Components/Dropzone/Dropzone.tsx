'use client';

import type {
  FileWithPath,
  DropzoneProps as IMantineDropzoneProps,
} from '@mantine/dropzone';
import type { MouseEvent, ReactElement } from 'react';

import { ActionIcon, Text, Tooltip } from '@mantine/core';
import { Dropzone as MantineDropzone } from '@mantine/dropzone';
import { Eye, Plus, X } from '@phosphor-icons/react';
import { BitConverter } from '@smile/haring-react';

import classes from './Dropzone.module.css';

export interface IFile extends Partial<FileWithPath> {
  name: string;
}

export interface IDropzoneProps
  extends Omit<IMantineDropzoneProps, 'children'> {
  browseLabel?: string;
  buttonPlusTitle?: string;
  children?: ReactElement;
  dragLabel?: string;
  files?: IFile[];
  onRemoveFile?: (file: IFile) => void;
}

export function Dropzone(props: IDropzoneProps): ReactElement {
  const {
    children,
    dragLabel = 'Drag and drop your documents here',
    buttonPlusTitle = 'Add button',
    browseLabel = 'Browse your device',
    files = [],
    onRemoveFile,
    ...MantineDropzoneProps
  } = props;

  function handleFileClick(e: MouseEvent<HTMLDivElement>): void {
    e.preventDefault();
    e.stopPropagation();
  }

  return (
    <MantineDropzone
      {...MantineDropzoneProps}
      classNames={{
        inner: classes.dropzoneInner,
        root: classes.dropzoneRoot,
      }}
    >
      <ActionIcon
        className={classes.buttonPlus}
        radius="xl"
        size="xl"
        title={buttonPlusTitle}
        variant="filled"
      >
        <Plus size={20} weight="bold" />
      </ActionIcon>
      <p>{dragLabel}</p>
      <p className={classes.dropzoneBrowse}>
        <Eye className={classes.eye} size={16} weight="bold" />
        {browseLabel}
      </p>
      <div className={classes.cardsFile}>
        {files.map((file, i) => {
          const { name, size } = file;
          return (
            <Tooltip
              key={`${i + i}`}
              label={name}
              position="bottom"
              radius={6}
              withArrow
            >
              <div
                aria-hidden="true"
                className={classes.cardFile}
                onClick={(e) => {
                  handleFileClick(e);
                }}
              >
                {Boolean(onRemoveFile) && (
                  <ActionIcon
                    className={classes.cardFileButtonClose}
                    onClick={() => onRemoveFile?.(file)}
                    radius="xl"
                    size="sm"
                    variant="filled"
                  >
                    <X size={8} weight="bold" />
                  </ActionIcon>
                )}
                <Text className={classes.cardFileText} span truncate>
                  {name}
                </Text>
                {size !== undefined && (
                  <span className={classes.cardFileText}>
                    <BitConverter options={{ locale: 'fr' }} value={size} />
                  </span>
                )}
              </div>
            </Tooltip>
          );
        })}
      </div>
      {children}
    </MantineDropzone>
  );
}
