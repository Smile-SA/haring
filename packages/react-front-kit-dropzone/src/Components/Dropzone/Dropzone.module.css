.buttonPlus {
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: center;
  margin: auto;
}

.cardFile {
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  height: 70px;
  margin: 10px;
  padding: 10px;
  position: relative;
  width: 80px;

  @mixin light {
    background: var(--mantine-color-gray-2);
  }

  @mixin dark {
    background: var(--mantine-color-black);
  }

  @mixin larger-than $mantine-breakpoint-md {
    background: red !important;
  }
}

.cardFileButtonClose {
  position: absolute;
  right: -7px;
  top: -7px;
}

.cardFileText {
  display: inline-block;
  font-size: 12px;
  height: fit-content;
  margin: auto;
  width: 100%;
}

.cardFileText:first-of-type {
  font-weight: 600;
}

.cardsFile {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.dropzoneBrowse {
  align-items: center;
  display: flex;
  font-weight: 600;
  justify-content: center;
  margin-top: 0px;
}

.dropzoneInner {
  margin: auto;
  pointer-events: auto;
  text-align: center;
}

.dropzoneInner p {
  margin: 10px;
}

.dropzoneRoot {
  display: flex;
  min-height: 100%;
  min-width: 100%;
}

.eye {
  margin-right: 8px;
}
