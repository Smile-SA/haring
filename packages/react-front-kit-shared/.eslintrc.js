module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
