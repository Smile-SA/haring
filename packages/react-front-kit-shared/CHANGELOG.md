# @smile/react-front-kit-shared

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/react-front-kit/pull/53) [`01ee3ea`](https://github.com/Smile-SA/react-front-kit/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
