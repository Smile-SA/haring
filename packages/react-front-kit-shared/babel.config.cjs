const { babelConfig } = require('test');

module.exports = babelConfig;
