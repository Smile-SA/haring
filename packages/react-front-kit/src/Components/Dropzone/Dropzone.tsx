'use client';

import type { DropzoneProps as IMantineDropzoneProps } from '@mantine/dropzone';
import type { ReactElement } from 'react';

import { ActionIcon } from '@mantine/core';
import { Dropzone as MantineDropzone } from '@mantine/dropzone';
import { createStyles } from '@mantine/styles';
import { Eye, Plus } from '@phosphor-icons/react';

export type IDropzoneProps = Omit<IMantineDropzoneProps, 'children'>;

const useStyles = createStyles(() => ({
  buttonPlus: {
    alignItems: 'center',
    cursor: 'pointer',
    display: 'flex',
    justifyContent: 'center',
    margin: 'auto',
  },
  dropzoneBrowse: {
    alignItems: 'center',
    display: 'flex',
    fontWeight: 600,
    justifyContent: 'center',
    marginTop: '0px',
  },
  dropzoneInner: {
    margin: 'auto',
    p: {
      margin: '10px',
    },
    textAlign: 'center',
  },
  dropzoneRoot: {
    display: 'flex',
    minHeight: '100%',
    minWidth: '100%',
  },
  eye: { marginRight: '8px' },
}));

export function Dropzone(props: IDropzoneProps): ReactElement {
  const { ...MantineDropzoneProps } = props;
  const { classes } = useStyles();
  return (
    <MantineDropzone
      {...MantineDropzoneProps}
      classNames={{
        inner: classes.dropzoneInner,
        root: classes.dropzoneRoot,
      }}
    >
      <ActionIcon
        className={classes.buttonPlus}
        radius="xl"
        size="xl"
        variant="filled"
      >
        <Plus size={20} weight="bold" />
      </ActionIcon>
      <p>Drag and drop your documents here</p>
      <p className={classes.dropzoneBrowse}>
        <Eye className={classes.eye} size={16} weight="bold" />
        Browse your device
      </p>
    </MantineDropzone>
  );
}
