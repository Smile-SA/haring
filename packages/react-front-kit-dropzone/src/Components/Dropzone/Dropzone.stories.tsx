import type { Meta, StoryObj } from '@storybook/react';

import { useStorybookArgsConnect } from '@smile/haring-react-shared/storybook-utils';

import { Dropzone as Cmp } from './Dropzone';

const meta = {
  argTypes: {
    browseLabel: {
      control: 'text',
    },
    dragLabel: {
      control: 'text',
    },
  },
  component: Cmp,
  decorators: [
    function Component(Story, ctx) {
      const args = useStorybookArgsConnect(ctx.args, {
        onDrop: 'files',
        onRemoveFile: (file) => ({
          files: ctx.args.files?.filter((f) => f !== file),
        }),
      });
      return <Story args={{ ...args }} />;
    },
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/Dropzone',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Dropzone: IStory = {
  args: {
    browseLabel: undefined,
    children: undefined,
    dragLabel: undefined,
    files: [],
  },
};
