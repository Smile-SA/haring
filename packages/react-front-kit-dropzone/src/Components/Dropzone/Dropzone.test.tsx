import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { Dropzone } from './Dropzone';

describe('Dropzone', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <Dropzone
        files={[
          {
            name: 'CERFA.postman_collection.json',
            size: 1579,
          },
          {
            name: 'CERFA.postman_collection.json',
            size: 1579,
          },
          {
            name: 'CERFA.postman_collection.json',
            size: 1579,
          },
          {
            name: 'CERFA.postman_collection.json',
            size: 1579,
          },
        ]}
        // eslint-disable-next-line no-console
        onDrop={(file) => console.log(file)}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
