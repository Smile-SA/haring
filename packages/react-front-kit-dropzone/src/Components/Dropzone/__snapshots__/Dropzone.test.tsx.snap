// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropzone matches snapshot 1`] = `
<div>
  <div
    class="mantine-Dropzone-root mantine-juj4dw"
    data-idle="true"
    role="presentation"
    tabindex="0"
  >
    <input
      multiple=""
      style="display: none;"
      tabindex="-1"
      type="file"
    />
    <div
      class="mantine-Dropzone-inner mantine-1th8yvx"
    >
      <button
        class="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-w4fu6v"
        title="Add button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 256 256"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M228,128a12,12,0,0,1-12,12H140v76a12,12,0,0,1-24,0V140H40a12,12,0,0,1,0-24h76V40a12,12,0,0,1,24,0v76h76A12,12,0,0,1,228,128Z"
          />
        </svg>
      </button>
      <p>
        Drag and drop your documents here
      </p>
      <p
        class="mantine-5jf51o"
      >
        <svg
          class="mantine-v0dvph"
          fill="currentColor"
          height="16"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M251,123.13c-.37-.81-9.13-20.26-28.48-39.61C196.63,57.67,164,44,128,44S59.37,57.67,33.51,83.52C14.16,102.87,5.4,122.32,5,123.13a12.08,12.08,0,0,0,0,9.75c.37.82,9.13,20.26,28.49,39.61C59.37,198.34,92,212,128,212s68.63-13.66,94.48-39.51c19.36-19.35,28.12-38.79,28.49-39.61A12.08,12.08,0,0,0,251,123.13Zm-46.06,33C183.47,177.27,157.59,188,128,188s-55.47-10.73-76.91-31.88A130.36,130.36,0,0,1,29.52,128,130.45,130.45,0,0,1,51.09,99.89C72.54,78.73,98.41,68,128,68s55.46,10.73,76.91,31.89A130.36,130.36,0,0,1,226.48,128,130.45,130.45,0,0,1,204.91,156.12ZM128,84a44,44,0,1,0,44,44A44.05,44.05,0,0,0,128,84Zm0,64a20,20,0,1,1,20-20A20,20,0,0,1,128,148Z"
          />
        </svg>
        Browse your device
      </p>
      <div
        class="mantine-kzfjfw"
      >
        <div
          aria-hidden="true"
          class="mantine-gcifvd"
        >
          <span
            class="mantine-1472iol"
          >
            <span
              class="mantine-1yjvws8"
            >
              CERFA.postman_collection.json
            </span>
          </span>
          <span
            class="mantine-1472iol"
          >
            1,58 kB
          </span>
        </div>
        <div
          aria-hidden="true"
          class="mantine-gcifvd"
        >
          <span
            class="mantine-1472iol"
          >
            <span
              class="mantine-1yjvws8"
            >
              CERFA.postman_collection.json
            </span>
          </span>
          <span
            class="mantine-1472iol"
          >
            1,58 kB
          </span>
        </div>
        <div
          aria-hidden="true"
          class="mantine-gcifvd"
        >
          <span
            class="mantine-1472iol"
          >
            <span
              class="mantine-1yjvws8"
            >
              CERFA.postman_collection.json
            </span>
          </span>
          <span
            class="mantine-1472iol"
          >
            1,58 kB
          </span>
        </div>
        <div
          aria-hidden="true"
          class="mantine-gcifvd"
        >
          <span
            class="mantine-1472iol"
          >
            <span
              class="mantine-1yjvws8"
            >
              CERFA.postman_collection.json
            </span>
          </span>
          <span
            class="mantine-1472iol"
          >
            1,58 kB
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
